src/txmempool.cpp

```diff
@@ -334,6 +335,21 @@ CTxMemPool::CTxMemPool(CBlockPolicyEstimator* estimator) :
     nCheckFrequency = 0;
 }
 
+void CTxMemPool::RegisterSignals() {
+    NotifyEntryRemoved.connect(boost::bind(&CTxMemPool::MempoolEntryRemoved, this, _1, _2));
+}
+
+void CTxMemPool::UnregisterSignals() {
+    NotifyEntryRemoved.disconnect(boost::bind(&CTxMemPool::MempoolEntryRemoved, this, _1, _2));
+}
+
+void CTxMemPool::MempoolEntryRemoved(CTransactionRef ptx, MemPoolRemovalReason reason)
+{
+    if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
+        GetMainSignals().TransactionRemovedFromMempool(ptx);
```

Comment:
Could `ptx` be moved to avoid unnecessary copies?

Reply:
Thanks @practicalswift. Good point. When `ptx` was passed by value I suspect we could have done that. However, per your suggestion https://github.com/bitcoin/bitcoin/pull/14384#discussion_r222600988 `ptx` is now passed by reference.